libnmstate/nm/nmclient.py

```diff
@@ -18,87 +18,14 @@
 #
 from collections import deque
 from contextlib import contextmanager
-import functools
 import logging
-import sys
-
-import gi
-
-try:
-    gi.require_version("NM", "1.0")  # NOQA: F402
-    from gi.repository import NM  # pylint: disable=no-name-in-module
-except ValueError:
-    NM = None
-
-from gi.repository import Gio
-from gi.repository import GLib
-from gi.repository import GObject
 
 from libnmstate import error
 
-GObject
+from .common import GLib
+from .common import Gio
 
 _mainloop = None
-_nmclient = None
-
-NM_MANAGER_ERROR_DOMAIN = "nm-manager-error-quark"
-_NMCLIENT_CLEANUP_TIMEOUT = 5
-
-
-def nm_version():
-    return NM.Client.get_version(client())
-
-
-def _delete_client():
-    global _nmclient
-    if _nmclient:
-        _nmclient.delete_client()
-        _nmclient = None
-
-
-def client():
-    global _nmclient
-    if _nmclient is None:
-        if NM:
-            _nmclient = _NmClient()
-            if not _nmclient.client.get_nm_running():
-                logging.error(
-                    "NetworkManager is not running, please make sure"
-                    "it is installed and running prior to running nmstate.\n"
-                    "Check the documentation for more information."
-                )
-                raise error.NmstateDependencyError(
-                    "NetworkManager is not running"
-                )
-        else:
-            logging.error(
-                "Missing introspection data for libnm"
-                "please make sure to install it prior to running nmstate.\n"
-                "Check the documentation for more information."
-            )
-            raise error.NmstateDependencyError(
-                "Missing introspection data for libnm"
-            )
-    return _nmclient.client
-
-
-def nmclient_context(func):
-    """
-    Decorator for handling NM.Client creation and clean up
-    """
-
-    @functools.wraps(func)
-    def wrapped(*args, **kwargs):
-        try:
-            _delete_client()
-            # Delete NM.Client is required, if not, previous changed interfaces
-            # might not updated in current NM.Client caches.
-            ret = func(*args, **kwargs)
-        finally:
-            _delete_client()
-        return ret
-
-    return wrapped
 
 
 def mainloop(refresh=False):
```

Comment:
Seems that this is the only method left here.
Therefore, this module name is no longer correct. It now represents only the `mainloop`.
Can you change its name? (you can do it in a following commit, to avoid rebase conflict mess)

Reply:
That file will be removed in PR #362 
Can we keep it here to save the rebase mess?

Reply:
Never mind. Already a rebase mess. Renamed to mainloop.